### PR TITLE
migrate to `esConsumes` several test codes in the SiStrip area

### LINKED ISSUE #51: More SiStrips ES get() calls
* `CalibTracker/SiStripESProducers`
```
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/f0e520f39471b6d35db786db8917f45a/opt/cmssw/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_2_CMSDEPRECATED_X_2021-11-03-2300/src/CalibTracker/SiStripESProducers/test/testSiStripQualityESProducer.cc:33:69: warning: 'bool edm::eventsetup::EventSetupRecord::get(const string&, HolderT&) const [with HolderT = edm::ESHandle<SiStripQuality>; std::string = std::__cxx11::basic_string<char>]' is deprecated [-Wdeprecated-declarations]
    33 |   iSetup.get<SiStripQualityRcd>().get(dataLabel_, SiStripQualityESH_);
```
and
```
 /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/f0e520f39471b6d35db786db8917f45a/opt/cmssw/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_2_CMSDEPRECATED_X_2021-11-03-2300/src/CalibTracker/SiStripESProducers/test/testSiStripQualityESProducer.cc:43:77: warning: 'bool edm::eventsetup::EventSetupRecord::get(const string&, HolderT&) const [with HolderT = edm::ESHandle<SiStripQuality>; std::string = std::__cxx11::basic_string<char>]' is deprecated [-Wdeprecated-declarations]
    43 |     iSetup.get<SiStripQualityRcd>().get(dataLabelTwo_, twoSiStripQualityESH_);
```
and
```
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/f0e520f39471b6d35db786db8917f45a/opt/cmssw/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_2_CMSDEPRECATED_X_2021-11-03-2300/src/CalibTracker/SiStripESProducers/test/CheckPhase2Cabling.cc:94:58: warning: 'bool edm::eventsetup::EventSetupRecord::get(HolderT&) const [with HolderT = edm::ESHandle<Phase2TrackerCabling>]' is deprecated [-Wdeprecated-declarations]
    94 |   iSetup.get<Phase2TrackerCablingRcd>().get(cablingHandle);
```
then
 * `CalibTracker/SiStripDCS`
```
 /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/f0e520f39471b6d35db786db8917f45a/opt/cmssw/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_2_CMSDEPRECATED_X_2021-11-03-2300/src/CalibTracker/SiStripDCS/test/Synchronization/SyncDCSO2O.cc:92:46: warning: 'bool edm::eventsetup::EventSetupRecord::get(HolderT&) const [with HolderT = edm::ESHandle<SiStripDetVOff>]' is deprecated [-Wdeprecated-declarations]
    92 |   iSetup.get<SiStripDetVOffRcd>().get(detVOff);
```
and
  * `CalibFormats/SiStripObjects`
```
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/f0e520f39471b6d35db786db8917f45a/opt/cmssw/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_2_CMSDEPRECATED_X_2021-11-03-2300/src/CalibFormats/SiStripObjects/test/plugins/testSiStripHashedDetId.cc:46:50: warning: 'bool edm::eventsetup::EventSetupRecord::get(HolderT&) const [with HolderT = edm::ESHandle<TrackerGeometry>]' is deprecated [-Wdeprecated-declarations]
    46 |   setup.get<TrackerDigiGeometryRecord>().get(geom);
```
and
 * `CondFormats/PhysicsToolsObjects`
```
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/f0e520f39471b6d35db786db8917f45a/opt/cmssw/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_2_CMSDEPRECATED_X_2021-11-03-2300/src/CondFormats/PhysicsToolsObjects/test/SiStripDeDx2DReader.cc:41:60: warning: 'bool edm::eventsetup::EventSetupRecord::get(HolderT&) const [with HolderT = edm::ESHandle<PhysicsTools::Calibration::VHistogramD2D>]' is deprecated [-Wdeprecated-declarations]
    41 |   iSetup.get<SiStripDeDxProton_2D_Rcd>().get(SiStripDeDx2D_);
```
and
```
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/f0e520f39471b6d35db786db8917f45a/opt/cmssw/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_2_CMSDEPRECATED_X_2021-11-03-2300/src/CondFormats/PhysicsToolsObjects/test/SiStripDeDx3DReader.cc:41:60: warning: 'bool edm::eventsetup::EventSetupRecord::get(HolderT&) const [with HolderT = edm::ESHandle<PhysicsTools::Calibration::Histogram3D<double> >]' is deprecated [-Wdeprecated-declarations]
    41 |   iSetup.get<SiStripDeDxProton_3D_Rcd>().get(SiStripDeDx3D_);
```
and
```
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/f0e520f39471b6d35db786db8917f45a/opt/cmssw/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_2_CMSDEPRECATED_X_2021-11-03-2300/src/CondFormats/PhysicsToolsObjects/test/SiStripDeDxMipReader.cc:19:54: warning: 'bool edm::eventsetup::EventSetupRecord::get(HolderT&) const [with HolderT = edm::ESHandle<PhysicsTools::Calibration::Histogram2D<double> >]' is deprecated [-Wdeprecated-declarations]
    19 |   iSetup.get<SiStripDeDxMipRcd>().get(SiStripDeDxMip_);
```

### PR DESCRIPTION
**Pull request by mmusich**
resolves https://github.com/cms-AlCaDB/AlCaTools/issues/51

#### PR description:

In response to https://github.com/cms-AlCaDB/AlCaTools/issues/51.
Profited to modernize the code.

#### PR validation:

`cmssw` compiles

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A
cc:
@robervalwalsh

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please test

File: CalibFormats/SiStripObjects/test/plugins/testSiStripHashedDetId.cc [hunk 1]
```diff
@@ -12,18 +21,30 @@
...
+public:
+  testSiStripHashedDetId(const edm::ParameterSet &);
+  ~testSiStripHashedDetId();
+
+  void initialize(edm::EventSetup const &);
+  void analyze(const edm::Event &, const edm::EventSetup &);
+
+private:
+  //-------------- member data ----------------//
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
```
Comment (mmusich) [inline CalibFormats/SiStripObjects/test/plugins/testSiStripHashedDetId.cc:40]:
```suggestion
  const  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
```

File: CalibTracker/SiStripESProducers/test/CheckPhase2Cabling.cc [hunk 2]
```diff
@@ -40,48 +40,26 @@
...
   virtual void analyze(const edm::Event&, const edm::EventSetup&);
-  virtual void endJob();
-
-  //virtual void beginRun(edm::Run const&, edm::EventSetup const&);
-  //virtual void endRun(edm::Run const&, edm::EventSetup const&);
-  //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&);
-  //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&);
 
   // ----------member data ---------------------------
+  edm::ESGetToken<Phase2TrackerCabling, Phase2TrackerCablingRcd> cablingToken_;
```
Comment (mmusich) [inline CalibTracker/SiStripESProducers/test/CheckPhase2Cabling.cc:54]:
```suggestion
  const edm::ESGetToken<Phase2TrackerCabling, Phase2TrackerCablingRcd> cablingToken_;
```

File: CalibTracker/SiStripESProducers/test/testSiStripQualityESProducer.cc [hunk 3]
```diff
@@ -1,20 +1,46 @@
...
+  explicit testSiStripQualityESProducer(const edm::ParameterSet&);
+  ~testSiStripQualityESProducer() = default;
+
+  void analyze(const edm::Event&, const edm::EventSetup&);
+
+private:
+  void printObject(const SiStripQuality*);
+
+  edm::ESGetToken<SiStripQuality, SiStripQualityRcd> qualityToken_;
+  edm::ESGetToken<SiStripQuality, SiStripQualityRcd> qualityTokenTwo_;
```
Comment (mmusich) [inline CalibTracker/SiStripESProducers/test/testSiStripQualityESProducer.cc:29]:
```suggestion
  const edm::ESGetToken<SiStripQuality, SiStripQualityRcd> qualityToken_;
  const edm::ESGetToken<SiStripQuality, SiStripQualityRcd> qualityTokenTwo_;
```

File: CondFormats/PhysicsToolsObjects/test/SiStripDeDx2DReader.cc [hunk 4]
```diff
@@ -20,25 +20,23 @@
...
+class SiStripDeDx2DReader : public edm::one::EDAnalyzer<> {
 public:
   explicit SiStripDeDx2DReader(const edm::ParameterSet&);
   ~SiStripDeDx2DReader();
 
   void analyze(const edm::Event&, const edm::EventSetup&);
 
 private:
-  //  uint32_t printdebug_;
+  edm::ESGetToken<PhysicsTools::Calibration::VHistogramD2D, SiStripDeDxProton_2D_Rcd> SiStripDeDx2DToken_;
```
Comment (mmusich) [inline CondFormats/PhysicsToolsObjects/test/SiStripDeDx2DReader.cc:31]:
```suggestion
  const edm::ESGetToken<PhysicsTools::Calibration::VHistogramD2D, SiStripDeDxProton_2D_Rcd> SiStripDeDx2DToken_;
```

File: CondFormats/PhysicsToolsObjects/test/SiStripDeDx3DReader.cc [hunk 5]
```diff
@@ -20,25 +20,25 @@
...
-class SiStripDeDx3DReader : public edm::EDAnalyzer {
+class SiStripDeDx3DReader : public edm::one::EDAnalyzer<> {
 public:
   explicit SiStripDeDx3DReader(const edm::ParameterSet&);
   ~SiStripDeDx3DReader();
 
   void analyze(const edm::Event&, const edm::EventSetup&);
 
 private:
+  edm::ESGetToken<PhysicsTools::Calibration::HistogramD3D, SiStripDeDxProton_3D_Rcd> SiStripDeDx3DToken_;
```
Comment (mmusich) [inline CondFormats/PhysicsToolsObjects/test/SiStripDeDx3DReader.cc:31]:
```suggestion
  const edm::ESGetToken<PhysicsTools::Calibration::HistogramD3D, SiStripDeDxProton_3D_Rcd> SiStripDeDx3DToken_;
```

File: CondFormats/PhysicsToolsObjects/test/SiStripDeDxMipReader.cc [hunk 6]
```diff
@@ -1,47 +1,46 @@
...
+
+class SiStripDeDxMipReader : public edm::one::EDAnalyzer<> {
+public:
+  explicit SiStripDeDxMipReader(const edm::ParameterSet&);
+  ~SiStripDeDxMipReader();
+
+  void analyze(const edm::Event&, const edm::EventSetup&);
+
+private:
+  edm::ESGetToken<PhysicsTools::Calibration::HistogramD2D, SiStripDeDxMipRcd> SiStripDeDxMipToken_;
```
Comment (mmusich) [inline CondFormats/PhysicsToolsObjects/test/SiStripDeDxMipReader.cc:26]:
```suggestion
  const edm::ESGetToken<PhysicsTools::Calibration::HistogramD2D, SiStripDeDxMipRcd> SiStripDeDxMipToken_;
```

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
+1
 * Jenkins test pass, only MsgLogger warnings
 * SA warnings are not connected to the purpose of this PR

**Comment by qliphy** [comment]
+1

**Pull request merged**